protocols/v2/binary-sv2/serde-sv2/src/primitives/sequences/mod.rs

```diff
@@ -22,6 +23,23 @@ struct Seq<'s, T: Clone + Serialize + TryFromBSlice<'s>> {
     _a: core::marker::PhantomData<T>,
 }
 
+impl<'s, T: Clone + Serialize + TryFromBSlice<'s>> Seq<'s, T> {
+    pub fn parse(self) -> Result<Vec<T>, ()> {
+        let mut ret = Vec::with_capacity(self.size as usize);
+        let elem_len = self.size as usize;
+        let mut first = 0;
+        let mut last = elem_len;
+        while last <= self.data.len() {
+            let slice = &self.data[first..last];
+            let elem = T::try_from_slice(slice).map_err(|_| ())?;
```

Comment:
Should we add an error variant here?

Reply:
An `crate::error::Error::PrimitiveConversionError` should be the right solution, what do you think? 
BTW this part of code is used only in the message generator and not for production.

Reply:
Ah, we can leave as is then

Reply:
Having second thoughts about this. As it is exposed through the public API, it might make more sense to handle errors properly here.
Also, I have a small question:
If `self.size` is bigger than `self.data.len`, is returning empty array with `self.size` cap(the current behavior) makes sense or an error should be returned?